### PR TITLE
Allow `cmd` fields in `tool.pdm.scripts` section to contain arrays in addition to strings

### PR DESCRIPTION
**Pull request by ntninja**
## Pull Request Check List

- [X] A news fragment is added in `news/` describing what is new.
- [X] Test cases added for changed code.

## Describe what you have changed in this PR.

Field `cmd` in `tools.pdm.scripts` configuration items now allows specifying an argument array instead of a string.

### CODE REVIEW

**Review by frostming — CHANGES_REQUESTED** [review]
Please follow the output of linters to fix the code style.

File: pdm/cli/commands/run.py [hunk 1]
```diff
@@ -84,9 +84,9 @@ def _run_command(
 
     def _normalize_script(self, script):
         if not getattr(script, "items", None):
-            # Must be a string, regard as the same as {cmd = "..."}
+            # Regard as the same as {cmd = … }
```
Comment (frostming) [inline pdm/cli/commands/run.py:87]: Can you change the non-ASCII `…` to three dots?

**Review by frostming — APPROVED** [review]

**Comment by frostming** [comment]
Thank you for contributing

**Comment by ntninja** [comment]
Thanks for having me!

**Pull request merged**